akka-testkit-typed/src/main/scala/akka/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -0,0 +1,106 @@
+package akka.testkit.typed.scaladsl
+
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.actor.typed.{ ActorRef, ActorSystem, Behavior, Props }
+import akka.annotation.{ ApiMayChange, InternalApi }
+import akka.testkit.typed.TestKitSettings
+import akka.testkit.typed.internal.{ ActorTestKitGuardian, TestKitUtils }
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object ActorTestKit {
+
+  def shutdown(
+    system:               ActorSystem[_],
+    duration:             Duration,
+    verifySystemShutdown: Boolean        = false): Unit =
+    TestKitUtils.shutdown(system, duration, verifySystemShutdown)
+
+}
+
+/**
+ * Testkit for asynchronous testing of typed actors, meant for mixing into the test class.
+ *
+ * Provides a typed actor system started on creation, used for all test cases and shut down when `shutdown` is called.
+ *
+ * The actor system has a custom guardian that allows for spawning arbitrary actors using the `spawn` methods.
+ *
+ * Designed to work with any test framework, but framework glue code that calls shutdown after all tests has
+ * run needs to be provided by the user.
+ *
+ * For synchronous testing of a `Behavior` see [[BehaviorTestKit]]
+ */
+@ApiMayChange
+trait ActorTestKit {
+  /**
+   * Actor system name based on the test it is mixed into, override to customize, or pass to constructor
+   * if using [[ActorTestKit]] rather than [[ActorTestKit]]
+   */
+  protected def name: String = TestKitUtils.testNameFromCallStack(classOf[ActorTestKit])
+
+  /**
+   * Configuration the actor system is created with, override to customize, or pass to constructor
+   * if using [[ActorTestKit]] rather than [[ActorTestKit]]
+   */
+  def config: Config = ConfigFactory.empty()
+
+  /**
+   * TestKit settings used in the tests, override or provide custom config to customize
+   */
+  protected implicit def testkitSettings = TestKitSettings(system)
+
+  private val internalSystem: ActorSystem[ActorTestKitGuardian.TestKitCommand] =
+    if (config == ConfigFactory.empty) ActorSystem(ActorTestKitGuardian.testKitGuardian, name)
```

Comment:
If the `config` is not defined this will load `application.conf`. I'm on the fence, but that might be an alright default also for tests (even though I personally would never use it like that). Anyway it should be clearly documented.

It might be slightly strange since `ActorSystem(guardian, "Sys", config = Some(ConfigFactory.empty))` doesn't have the same semantics. Perhaps make it `def config: Option[Config] = None`?

Reply:
I started with an option but it felt messy for the tests, just overloading and providing config is so much nicer I think it is worth it. What about providing some other minimal placeholder testkit-specific config instance to compare with?

Reply:
yes, such placeholder (comparing with `eq`) would be fine